### PR TITLE
Front end Random Message

### PR DESCRIPTION
**Pull request by Bishoy-Samwel**
Users can extract the random messages from API 
[Back-end](https://github.com/Bishoy-Samwel/Rails-React-Intro)

### CODE REVIEW

File: src/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { applyMiddleware, createStore } from 'redux';
+import { composeWithDevTools } from 'redux-devtools-extension';
+import thunk from 'redux-thunk';
+
+const initialState = {
+  msg: { title: 'Hello', body: 'English Message' },
+};
+
+const rootReducer = (state = initialState, action) => {
+  console.log(action.type, 'request');
```
Comment (bimbolabuari) [inline src/configureStore.js]: [Optional] Kindly remove all console logs from your code, to make it look more professional ✊

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @Bishoy-Samwel,

You have done a good job so far 🚀  

## Highlights 🎉 
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Your PR provide a link to the Rails API (backend app)


There are some issues that you still need to work on to go to the next project but you are almost there! 🎉 

## Required Changes ♻️
- As required, Let's create an `app` component with react-router

>image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

--------


_Reviewed by  @bimbolabuari_

File: src/index.js [hunk 2]
```diff
@@ -1,17 +1,49 @@
...
+              <li>
+                <Link to="/randomMessage">Random Message</Link>
+              </li>
+            </ul>
+          </nav>
+
+          <Routes>
+            <Route path="/" element={<Home />} />
+            <Route path="/randomMessage" element={<Message msg="hi" />} />
+          </Routes>
```
Comment (oloomoses) [inline src/index.js]: - [x] Routing should be done in the `App` component and not `index.js`. Kindly create an App component as suggested in the previous review and do the routing from there.

File: src/configureStore.js (hunk 1 above)
Comment (oloomoses) [reply src/configureStore.js]: - [x] Kindly remove the console log statement as suggested. It is ugly and exposes your data.

File: src/message.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+};
+
+//thunk
+const getRandomMsg = () => (dispacth) => {
+  dispacth({ type: GET_RANDOM_MSG_REQUEST });
+  return fetch('http://[::1]:4000/api/v1/randomMessage.json')
+    .then((response) => response.json())
+    .then((json) => dispacth(getRandomMsgSuccess(json)))
+    .catch((error) => console.log(error));
+};
```
Comment (oloomoses) [inline src/message.jsx]: - [x] Kindly remove the console log statement as suggested. It is ugly and exposes your data.

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi @Bishoy-Samwel, 

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly fix the linter failing check before submitting for another review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @Bishoy-Samwel, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**